instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integration/rum/FragmentTrackingTest.kt

```diff
@@ -73,8 +74,8 @@ internal abstract class FragmentTrackingTest :
         // swipe to close the view
         onView(ViewMatchers.withId(R.id.btn_last)).perform(ViewActions.click())
         instrumentation.waitForIdleSync()
-        Thread.sleep(200) // give time to the view id to update
-
+        Thread.sleep(200)
```

Comment:
> Tiny remark 
> Should we remove this sleep since we have `waitForPendingRUMEvents` below?

Reply:
I want to keep it just for weird low API devices. To make sure it has time to stabilise. For some reason sometimes `instrumentation.waitForIdleSync` doesn't work really well on those emulators.

Reply:
I think @ambushwork is asking here about a bit different method - `waitForPendingRUMEvents`.

But speaking about `waitForIdleSync` - I'm curious why it doesn't work. Description of `IdleHandler` [says](https://cs.android.com/android/platform/superproject/main/+/main:frameworks/base/core/java/android/os/MessageQueue.java;l=1131-1139?q=MessageQueue):

```
/**
 * Called when the message queue has run out of messages and will now
 * wait for more.  Return true to keep your idle handler active, false
 * to have it removed.  This may be called if there are still messages
 * pending in the queue, but they are all scheduled to be dispatched
 * after the current time.
 */
```

So, if click event above was added to the queue before the `waitForIdleSync`, it should work just fine? Or we have something in the test structure that works in async way?

Reply:
I mean I can't tell you why it is not really working. It is just based on my previous experience with the emulators. I don't really mind removing this and hope this will work on all the emulators. I am about do it now actually.